### ISSUE TITLE
Question: Cross-platform demo anywhere (iOS, Android)

### ISSUE THREAD

**Issue by dgerding** [SciSharp/TensorFlow.NET#597]
Am I correct in assuming I can use TensorFlow.NET and target iOS, Android, Windows? Assuming key issues is matching native tensor flow assembly with target?

Any examples or references appreciated.

**Comment by Oceania2018** [comment]
@dgerding We don't have demo yet, as far as I know, you can use Xamaria and Tensorflow Lite to deploy tensorflow model in mobile.

**Comment by dgerding** [comment]
Thanks for the reply. If I find the time I'll see if I can put something together. Tensor flow lite looks cool but I think takes me back towards platform-specific projects which I can avoid a lot of with Xamarin. I didn't know about it and will read more. Thanks again.

**Comment by MannyKayy** [comment]
I strongly agree, A cross platform mobile demo with Xamarin would be a really useful addition !

**Comment by kerryjiang** [comment]
The mobile demo would be great. But should all the AI computation happen at the server side?

**Comment by MannyKayy** [comment]
I think all the computation should be on device.. It would be handy to have any one of these [examples](https://www.tensorflow.org/lite/examples) as a demo.

**Comment by dgerding** [comment]
Here's what I am really after: using TensorFlow based tools like BERT, primarily for inference but, in theory, for learning too. Am I wrong in thinking TensorFlow.net "should" let me get to that outcome? I tried using Cherub.Net but it looks like that project might be dormant.

I will happily build out and share a TensorFlow.net-based demo that targets cross-platform (droid, ios, windows?) but I need at least one example of using tensorflow.net with C# on either Android or iOS. I should be able to figure things out from there. I'll keep looking in mean time.

Thanks for a very promising project.

Dave Gerding

**Comment by Esther2013** [comment]
It will be good if you can do the research and push the code to [examples](https://github.com/SciSharp/SciSharp-Stack-Examples).

**Comment by dgerding** [comment]
@Esther2013 and all, if people are saying "to use TensorFlow Lite" are they really implying "use TensorFlow Lite models with TensorFlow.net code"? Confused about how Tensor Flow Lite fits in (or doesn't) with TensorFlow.net?

**Comment by dgerding** [comment]
Okay, TF Lite is inference only. Any other responses to questions above appreciated.

**Comment by MannyKayy** [comment]
If this helps, my specific use case is as follows: 
1. I would like to run an image classification model that would analyse user images on the device... 
2. the results of this analysis may possibly at some later time be used to train a simpler model on the device in order to make personalized predictions. 

Just getting step 1 to work with any of the examples mentioned would be very useful. Due to data privacy laws in the EU, this data is too sensitive to be offloaded to remote servers... everything would have to stay on device.

**Comment by BlueCode2019** [comment]
I recently encountered a demand for this cross-platform application, and I look forward to an example
